### PR TITLE
Add conditional if VPC should be created

### PR DESCRIPTION
**Pull request by akonchyts**
Sometimes there is a need to have a way to create VPC resources conditionally. Therefore a parameter create_vpc has been added to allow create or not VPC depending on the conditions

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -8,7 +8,7 @@ locals {
 }
 
 resource "aws_route53_zone" "main" {
-  count = var.create_route53_zone ? 1 : 0
+  count = var.create_vpc && var.create_route53_zone ? 1 : 0
```
Comment (ToROxI) [inline main.tf]: Why you want to control route35 zone creation with `create_vpc` variable?

File: main.tf [hunk 2]
```diff
@@ -110,18 +114,18 @@ resource "aws_default_security_group" "this" {
...
 
 resource "aws_instance" "nat" {
-  count = var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
+  count = var.create_vpc && var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
 
   ami                    = data.aws_ami.nat.id
   instance_type          = var.instance_type
   availability_zone      = element(var.availability_zones, count.index)
-  vpc_security_group_ids = [aws_default_security_group.this.id]
+  vpc_security_group_ids = [aws_default_security_group.this[count.index].id]
```
Comment (ToROxI) [inline main.tf]: looks like there will be an error in case if `local.ec2_nat_count` > 1

**Review by ToROxI — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (akonchyts) [reply main.tf]: Brought back with commit https://github.com/lean-delivery/tf-module-aws-core/pull/11/commits/02bfa3563c39c60528ac475e8610738609ba8019

File: main.tf (hunk 2 above)
Comment (akonchyts) [reply main.tf]: yes, you are right.
**vpc_security_group_ids = [aws_default_security_group.this[0].id]** would work as there is only one default security group in a VPC

Comment (akonchyts) [reply main.tf]: fixed in commit https://github.com/lean-delivery/tf-module-aws-core/pull/11/commits/7ffc8c85e47c23463014dc897cb79f88d2881894
**[aws_default_security_group.this[0].id] -> aws_default_security_group.this[*].id**

**Review by akonchyts — COMMENTED** [review]
@ToROxI please check again, I have fixed the issues

**Review by ToROxI — APPROVED** [review]

**Pull request merged**